imports/plugins/core/checkout/client/templates/checkout/checkout.html

```diff
@@ -25,7 +25,15 @@
       </div>
     </div>
   {{else}}
-    {{> emptyCheckoutCart}}
+    {{#if currentUser}}
```

Comment:
![image](https://user-images.githubusercontent.com/1203639/33300009-62d98436-d3ab-11e7-9fb4-5b84ae3b39f2.png)

While this does seem to work to remove the flash of "your cart is empty" I'm seeing a side effect of causing an "endless" spinner to be rendered for any logged in user who happens to be on the `cart/checkout` route with an empty cart.

Reply:
@spencern, @zenweasel 
The reason for this is happens is the way I implemented the checks. The user is shown with an infinite spinner which was a wrong implementation regarding the fact that every logged in user should be able to see the `empty cart page` if there are not items in the cart. 

#### Fix
- I created a helper method which checks if the items in the cart is available and if it has a value.

#### Test
- Login
- go to `cart/checkout` route.
- `empty cart error` should show now
- Checks works fine now. It register the user without flashing an empty cart